cas/cas.go

```diff
@@ -264,6 +307,17 @@ func (ds Store) WriteACI(r io.Reader, latest bool) (string, error) {
 // Users of treestore should call ds.RenderTreeStore before using it to ensure
 // that the treestore is completely rendered.
 func (ds Store) RenderTreeStore(key string, rebuild bool) error {
+	// this lock references the treestore dir for the specified key. This
+	// is different from a lock on an image key as internally
+	// treestore.Write calls the acirenderer functions that use GetACI and
+	// GetImageManifest which are taking the image lock.
+	treestoreLockKey := key + "-treestore"
```

Comment:
> we should have a function to generate this lock name instead of repeating it.

@philips Instead of doing this I switched using another lockdir for treestore locks. It seems more logical.